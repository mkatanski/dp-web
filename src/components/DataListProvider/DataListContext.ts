import { ReducerName } from "store";
import { createContextWithPartial } from "utils/context/createContextWithPartial";

export type DataListStateContextProps = {
  reducerName: ReducerName;
};

export type DataListActionContextProps = {
  onPaginationChange: (newOffset: number) => void;
};

export const DataListStateContext = createContextWithPartial<
  DataListStateContextProps
>({});

export const DataListActionContext = createContextWithPartial<
  DataListActionContextProps
>({});
