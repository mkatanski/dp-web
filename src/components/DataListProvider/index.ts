export * from "./DataListProvider";
export * from "./DataListContext";
