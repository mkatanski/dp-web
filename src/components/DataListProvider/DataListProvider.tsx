import React from "react";
import { ReducerName } from "store";
import { DataListStateContext, DataListActionContext } from "./DataListContext";

export type DataListProviderProps = React.PropsWithChildren<{
  onPaginationChange?: (newOffset: number) => void;
  reducerName: ReducerName;
}>;

export const DataListProvider: React.FC<DataListProviderProps> = ({
  children,
  reducerName,
  onPaginationChange = () => undefined
}: DataListProviderProps) => {
  return (
    <DataListStateContext.Provider value={{ reducerName }}>
      <DataListActionContext.Provider value={{ onPaginationChange }}>
        {children}
      </DataListActionContext.Provider>
    </DataListStateContext.Provider>
  );
};
