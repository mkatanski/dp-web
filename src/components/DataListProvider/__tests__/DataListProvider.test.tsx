import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

/* Uncomment below if needed */
// import { withThemeProvider, withDesktopSize } from 'testsHelpers';
// import { render, getByTestId } from "@testing-library/react";

import { DataListProvider } from "../DataListProvider";

/* Uncomment below if You need/want to use enzyme */
// import enzyme from "enzyme";

/* Component mock example */
// jest.mock("[module]", () => ({
//   [ComponentName]: ({children, ...rest}: React.PropsWithChildren<{}>) => (
//     <div id="[ComponentName]" {...rest}>{children}</div>
//   )
// }));

describe("components", () => {
  describe("DataListProvider", () => {
    it("should render DataListProvider component", () => {
      const tree = TestRenderer.create(<DataListProvider />);

      expect(tree).toMatchSnapshot();
    });
  });
});
