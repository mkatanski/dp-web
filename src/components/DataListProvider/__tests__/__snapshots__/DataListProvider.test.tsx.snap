// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components DataListProvider should render DataListProvider component 1`] = `null`;
